File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
        public int maxArea(int[] height) {
                int maxArea = 0;
                        int curArea = 0;
                                int left = 0;
                                        int right = height.length -1;
                                                while(left<right){
                                                            curArea = (right-left) * Math.min(height[right], height[left]);
                                                                        maxArea = Math.max(curArea, maxArea);
                                                                                    if(height[right] > height[left]){
                                                                                                    left++;
                                                                                                                }
                                                                                                                            else{
                                                                                                                                            right--;
                                                                                                                                                        }
                                                                                                                                                                }
                                                                                                                                                                        return maxArea;
                                                                                                                                                                            }
                                                                                                                                                                            }

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size(); 

        int start = 0; 
        int end = n - 1;

        int area = 0; 
        int result = 0; 
        
        while (start < end) {
            int breadth = end - start; // Calculate the distance (breadth) between the two pointers
            area = breadth * min(height[start], height[end]); // Calculate the area between the two lines
            result = max(area, result);

            // Move the pointer pointing to the shorter line towards the center
            if (height[start] <= height[end]) {
                start++;
            } else {
                end--;
            }
        } 

        return result; 
    }
};